src/libfsm/determinise_internal.h

```diff
@@ -98,26 +103,68 @@ struct mappingstack {
 #define DEF_OUTPUT_CEIL 4
 #define DEF_DST_CEIL 4
 
+/* Mark an ID as a cached result ID (1) rather than a state ID (0). */
+#define RESULT_BIT ((fsm_state_t)1U << (8U*sizeof(fsm_state_t) - 1))
```

Comment:
Strictly speaking, I don't think the `RESULT_BIT` flag is necessary. Iit's redundant, because in all cases we _should_ be able to distinguish between result or state IDs by whether the pair of IDs are the same -- they can only match for caching the result from a single state, otherwise they have to differ. Using the flag gives an extra integrity check and makes some of the debugging printouts clearer, and the time spent on or-ing / masking the flag is probably inconsequential, but I'm a bit uneasy with how it clutters some of the dataflow. What do you think?